src/pages/auth/SignUp/SignUpDoctor.tsx

```diff
@@ -31,69 +35,195 @@ const schema = object({
   surname: string().required(),
   city: string().required(),
   email: string().email().required(),
-  password:
-    string()
-    .matches(/^.*(?=.{8,})((?=.*[!@#$%^&*()\-_=+{};:,<.>]){1})(?=.*\d)((?=.*[a-z]){1})((?=.*[A-Z]){1}).*$/,
-      "Password must contain at least 8 characters, one uppercase, one number and one special case character")
+  password: string()
+    .matches(
+      /^.*(?=.{8,})((?=.*[!@#$%^&*()\-_=+{};:,<.>]){1})(?=.*\d)((?=.*[a-z]){1})((?=.*[A-Z]){1}).*$/,
+      "Password must contain at least 8 characters, one uppercase, one number and one special case character"
+    )
     .required("Please enter your password"),
 });
 
 export const SignUpDoctor = () => {
   const navigate = useNavigate();
-  const { register, handleSubmit, formState: { errors } } = useForm<FormData>({
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<FormData>({
     resolver: yupResolver(schema),
     defaultValues: {
-      category: '',
-      name: '',
-      surname: '',
-      city: '',
-      email: '',
-      password: '',
-    }
-  });
-
-  const mutation = useMutation((data: IDoctor) => {
-    return API.post("rs-doctors-back", "/doctors/add-doctor", {
-      body: data
-    });
-  }, {
-    onSuccess: (_, defaultValues) => {
-      navigate(`/auth/sign-up-confirmation?email=${defaultValues.email}`);
+      category: "",
+      name: "",
+      surname: "",
+      city: "",
+      email: "",
+      password: "",
     },
   });
 
-  const onSubmit = handleSubmit(async ({ email, password, name, surname, city, category }) => {
-    try {
-      const doctorData = await AuthService.signUp({ email, password, profile: "doctor" });
-      const username = doctorData.user.getUsername();
-      mutation.mutate({
-        id: doctorData.userSub,
-        name: name,
-        surname: surname,
-        category: category,
-        email: username,
-        city: city
+  const mutation = useMutation(
+    (data: IDoctor) => {
+      return API.post("rs-doctors-back", "/doctors/add-doctor", {
+        body: data,
       });
-    } catch (err) {
-      console.log(err);
+    },
+    {
+      onSuccess: (_, defaultValues) => {
+        navigate(`/auth/sign-up-confirmation?email=${defaultValues.email}`);
+      },
     }
-  });
+  );
+
+  const onSubmit = handleSubmit(
+    async ({ email, password, name, surname, city, category }) => {
+      try {
+        const doctorData = await AuthService.signUp({
+          email,
+          password,
+          profile: "doctor",
+        });
+        const username = doctorData.user.getUsername();
+        mutation.mutate({
+          id: doctorData.userSub,
+          name: name,
+          surname: surname,
+          category: category,
+          email: username,
+          city: city,
+        });
+      } catch (err) {
```

Comment:
Why do u use catch here? I suppose that you can use mutation.isError and if it's true, show error [Snackbar ](https://mui.com/material-ui/react-snackbar/#simple-snackbars)or Alert